docs/export_results.rst

```diff
@@ -0,0 +1,166 @@
+Export your results
+===================
+Taichi has functions that help you **export visual results to images or videos**. This tutorial demonstrates how to use them step by step.
+
+Export images
+-------------
+
+- There are two ways to export visual results of your program to images.
+- The first and easier way is to make use of ``ti.GUI``.
+- The second way is to call some Taichi functions such as ``ti.imwrite``.
+
+Export images using ``ti.GUI.show``
++++++++++++++++++++++++++++++++++++
+
+- ``ti.GUI.show(filename)`` can not only display the GUI canvas on your screen, but also save the image to your specified ``filename``.
+- Note that the format of the image is fully determined by the suffix of ``filename``.
+- Taichi now supports saving to ``png``, ``jpg``, and ``bmp`` formats.
+- We recommend using ``png`` format. For example:
+
+.. code-block:: python
+
+    import taichi as ti
+    import os
+
+    ti.init()
+
+    pixels = ti.var(ti.u8, shape=(512, 512, 3))
+
+    @ti.kernel
+    def paint():
+        for i, j, k in pixels:
+            pixels[i, j, k] = ti.random() * 255
+
+    iterations = 1000
+    gui = ti.GUI("Random pixels", res=512)
+
+    # mainloop
+    for i in range(iterations):
+        paint()
+        gui.set_image(pixels)
+
+        filename = f'frame_{i:05d}.png'   # create filename with suffix png
+        print(f'Frame {i} is recorded in {filename}')
+        gui.show(filename)  # export and show in GUI
+
+- After running the code above, you will get a series of images in the current folder.
+
+Export images using ``ti.imwrite``
+++++++++++++++++++++++++++++++++++
+
+To save images without creating a ``ti.GUI``, use ``ti.imwrite``. For example:
+
+    .. code-block:: python
+
+        import taichi as ti
+
+        ti.init()
+
+        pixels = ti.var(ti.u8, shape=(512, 512, 3))
```

Comment:
Can we inform that both `ti.var, ti.Vector` is available? And all `shape[2] in [1, 3, 4]` is available?